File: JAVA/Bike.java
public class Bike {
    private final int gearone = 1;
    private final  int geartwo = 2;
    private final int gearthree = 3;
    private final int gearfour = 4;

    private int gearstatus = 0;
    private boolean bikeStatus = false;
    private int speedStatus = 0;

    public boolean getBikeStatus(){
        return this.bikeStatus;
    }



    public int getSpeedStatus(){
        return this.speedStatus;
    }

    public Boolean bikeOn(){
        if(bikeStatus==false){
            this.bikeStatus =  true;
            return true;
        }
        return false;
    }
    public boolean bikeOff(){
        if(bikeStatus==true){
            this.bikeStatus = false;
            return false;
        }
        return true;
    }

    public int getGearstatus(){
        return gearstatus;
    }


    public void accelerate(){
        if(bikeStatus){
            if (speedStatus <= 20 && speedStatus >= 0) {
                if (speedStatus == 0) {
                    this.gearstatus += gearone;
                }
                this.speedStatus += gearone;

            } else if (speedStatus >= 21 && speedStatus <= 30){
                if (speedStatus == 21){
                    this.gearstatus += 1;
                }
                this.speedStatus += geartwo;

            } else if (speedStatus >= 31 && speedStatus <= 40) {

                if (speedStatus == 31){
                    this.gearstatus += 1;
                }
                    this.speedStatus += gearthree;
            } else if (speedStatus >= 41){
                if (speedStatus == 43||speedStatus == 41 || speedStatus == 42){
                    if(gearstatus == 3){
                        this.gearstatus += 1;
                    }
                }
                this.speedStatus += gearfour;
            }
        }
    }


    public void decelerate(){
        if(bikeStatus){
            if (speedStatus <= 20 && speedStatus > 0) {
                if (speedStatus == 1) {
                    this.gearstatus -= gearone;
                }
                if (gearstatus >= 0) {
                    this.speedStatus -= gearone;

                }
            } else if (speedStatus >= 21 && speedStatus <= 30){
                if (gearstatus == 2){
                    this.gearstatus -= 1;
                }
                this.speedStatus -= geartwo;

            } else if (speedStatus >= 31 && speedStatus <= 40) {

                if (gearstatus ==3){
                    this.gearstatus -= 1;
                }
                this.speedStatus -= gearthree;
            } else if (speedStatus >= 41){
                    if(gearstatus == 4) {
                        this.gearstatus -= 1;
                    }
                    this.speedStatus -= gearfour;
            }
        }
    }



}

File: PYTHON/bike.py
class Bike:
    def __init__(self):
        self.GEARONE = 1
        self.GEARTWO = 2
        self.GEARTHREE = 3
        self.GEARFOUR = 4


        self.gearstatus = 0
        self.bikeStatus = False
        self.speedStatus = 0

    def get_bike_status(self):
        return self.bikeStatus

    def get_speed_status(self):
        return self.speedStatus

    def bike_on(self):
        if not self.bikeStatus:
            self.bikeStatus = True
            return True
        return False

    def bike_off(self):
        if self.bikeStatus:
            self.bikeStatus = False
            return False
        return True

    def get_gear_status(self):
        return self.gearstatus

    def accelerate(self):
        if self.bikeStatus:
            if 0 <= self.speedStatus <= 20:
                if self.speedStatus == 0:
                    self.gearstatus += self.GEARONE
                self.speedStatus += self.GEARONE

            elif 21 <= self.speedStatus <= 30:
                if self.speedStatus == 21:
                    self.gearstatus += 1
                self.speedStatus += self.GEARTWO

            elif 31 <= self.speedStatus <= 40:
                if self.speedStatus == 31:
                    self.gearstatus += 1
                self.speedStatus += self.GEARTHREE

            elif self.speedStatus >= 41:
                if self.speedStatus in [41, 42, 43] and self.gearstatus == 3:
                    self.gearstatus += 1
                self.speedStatus += self.GEARFOUR

    def decelerate(self):
        if self.bikeStatus:
            if 0 < self.speedStatus <= 20:
                if self.speedStatus == 1:
                    self.gearstatus -= self.GEARONE
                if self.gearstatus >= 0:
                    self.speedStatus -= self.GEARONE

            elif 21 <= self.speedStatus <= 30:
                if self.gearstatus == 2:
                    self.gearstatus -= 1
                self.speedStatus -= self.GEARTWO

            elif 31 <= self.speedStatus <= 40:
                if self.gearstatus == 3:
                    self.gearstatus -= 1
                self.speedStatus -= self.GEARTHREE

            elif self.speedStatus >= 41:
                if self.gearstatus == 4:
                    self.gearstatus -= 1
                self.speedStatus -= self.GEARFOUR